frontend/src/buttons/DeleteButton/DeleteButton.tsx

```diff
@@ -27,13 +28,15 @@ export const DeleteButton: React.FC<Props> = ({ device }) => {
     tooltip = 'Device must be offline'
   }
   if (device.shared) {
-    disabled = true
-    tooltip = 'You must be the device owner'
+    disabled = false
+    deleteDevice  = false
+    tooltip = 'Leave Device'
+    warning = "Are you sure to leave this device?"
   }
 
   const onDelete = () => {
     if (window.confirm(warning)) {
-      devices.destroy(device)
+      deleteDevice ? devices.destroy(device) : devices.unShare(device)
```

Comment:
If you use `device.shared` here I don't think you need the `deleteDevice` variable?